include/evm_runtime/config_wrapper.hpp

```diff
@@ -0,0 +1,128 @@
+#pragma once
+#include <eosio/eosio.hpp>
+#include <evm_runtime/tables.hpp>
+
+namespace evm_runtime {
+
+using eosio::check;
+
+struct config_wrapper {
+
+    config_wrapper(eosio::name self) : _self(self), _config(self, self.value) {
+        _exists = _config.exists();
+        if(_exists) {
+            _cached_config = _config.get();
+        }
+    }
+
+    ~config_wrapper() {
+        flush();
+    }
+
+    void flush() {
+        if( _evm_version_promote_required ) {
+            promote_pending_evm_version();
+            set_dirty();
+        }
+        if(!is_dirty()) {
+            return;
+        }
+        _config.set(_cached_config, _self);
+        clear_dirty();
+        _exists = true;
+    }
+
+    bool exists() {
+        return _exists;
+    }
+
+    inline eosio::unsigned_int get_version()const { return _cached_config.version; }
+    void set_version(const eosio::unsigned_int version) { _cached_config.version = version; set_dirty(); }
+
+    inline uint64_t get_chainid()const { return _cached_config.chainid; }
+    void set_chainid(uint64_t chainid) { _cached_config.chainid = chainid; set_dirty(); }
+
+    inline const eosio::time_point_sec& get_genesis_time()const { return _cached_config.genesis_time; }
+    void set_genesis_time(eosio::time_point_sec genesis_time) { _cached_config.genesis_time = genesis_time; set_dirty(); }
+
+    inline const eosio::asset& get_ingress_bridge_fee()const { return _cached_config.ingress_bridge_fee; }
+    void set_ingress_bridge_fee(const eosio::asset& ingress_bridge_fee) { _cached_config.ingress_bridge_fee = ingress_bridge_fee; set_dirty(); }
+
+    inline uint64_t get_gas_price()const { return _cached_config.gas_price; }
+    void set_gas_price(uint64_t gas_price) { _cached_config.gas_price = gas_price; set_dirty(); }
+
+    inline uint32_t get_miner_cut()const { return _cached_config.miner_cut; }
+    void set_miner_cut(uint32_t miner_cut) { _cached_config.miner_cut = miner_cut; set_dirty(); }
+
+    inline uint32_t get_status()const { return _cached_config.status; }
+    void set_status(uint32_t status) { _cached_config.status = status; set_dirty(); }
+
+    inline uint64_t get_evm_version()const {
+        uint64_t current_version = 0;
+        _evm_version_promote_required = false;
```

Comment:
`_evm_version_promote_required` should not be modified when getting the version. This could cause the `config` singleton to be updated at the end of the action even when not desired (e.g. in an `exec` action).